packages/server/src/utilities/usageQuota.js

```diff
@@ -47,59 +11,47 @@ exports.Properties = {
   VIEW: "views",
   USER: "users",
   AUTOMATION: "automationRuns",
-}
-
-exports.getAPIKey = async appId => {
-  if (!env.USE_QUOTAS) {
-    return { apiKey: null }
-  }
-  return apiKeyTable.get({ primary: appId })
+  APPS: "apps",
+  EMAILS: "emails",
 }
 
 /**
- * Given a specified API key this will add to the usage object for the specified property.
- * @param {string} apiKey The API key which is to be updated.
+ * Given a specified tenantId this will add to the usage object for the specified property.
  * @param {string} property The property which is to be added to (within the nested usageQuota object).
  * @param {number} usage The amount (this can be negative) to adjust the number by.
  * @returns {Promise<void>} When this completes the API key will now be up to date - the quota period may have
  * also been reset after this call.
  */
-exports.update = async (apiKey, property, usage) => {
+exports.update = async (property, usage) => {
   if (!env.USE_QUOTAS) {
     return
   }
+
   try {
-    await apiKeyTable.update(buildUpdateParams(apiKey, property, usage))
-  } catch (err) {
-    // conditional check means the condition failed, need to check why
-    if (err.code === "ConditionalCheckFailedException") {
-      // get the API key so we can check it
-      const keyObj = await apiKeyTable.get({ primary: apiKey })
-      // the usage quota or usage limits didn't exist
-      if (keyObj && (keyObj.usageQuota == null || keyObj.usageLimits == null)) {
-        keyObj.usageQuota =
-          keyObj.usageQuota == null ? DEFAULT_USAGE : keyObj.usageQuota
-        keyObj.usageLimits =
-          keyObj.usageLimits == null ? DEFAULT_PLAN : keyObj.usageLimits
-        keyObj.quotaReset = getNewQuotaReset()
-        await apiKeyTable.put({ item: keyObj })
-        return
-      }
-      // we have in fact breached the reset period
-      else if (keyObj && keyObj.quotaReset <= Date.now()) {
-        // update the quota reset period and reset the values for all properties
-        keyObj.quotaReset = getNewQuotaReset()
-        for (let prop of Object.keys(keyObj.usageQuota)) {
-          if (prop === property) {
-            keyObj.usageQuota[prop] = usage > 0 ? usage : 0
-          } else {
-            keyObj.usageQuota[prop] = 0
-          }
-        }
-        await apiKeyTable.put({ item: keyObj })
-        return
+    const db = getGlobalDB()
+    const quota = await db.get("usage_quota")
```

Comment:
Also, just to add to this, what happens in the scenario of the doc not existing and a 404 being thrown? Its unlikely but if it occurred could block the user complete,y if it occurs might be good to have a fallback in the error catching for `err.status === 404` to create the doc and retry, just for safety

Reply:
The doc should always exist for org creation in cloud, but for our only existing tenant (us) it will need backfilled. It does probably make sense just in case 👍